service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyServiceHandler.java

```diff
@@ -108,16 +108,41 @@ public void initializeSession(IoSession session, ConnectFuture future) {
             HttpConnectSession connectSession = (HttpConnectSession) session;
             connectSession.setVersion(acceptSession.getVersion());
             connectSession.setMethod(acceptSession.getMethod());
-            String acceptPathInfo = acceptSession.getPathInfo().toString();
-            if (acceptPathInfo.startsWith("/")) {
-                acceptPathInfo = acceptPathInfo.substring(1);
+            URI connectURI = computeConnectPath(connectSession.getRequestURI());
+            connectSession.setRequestURI(connectURI);
+            processRequestHeaders(acceptSession, connectSession);
+        }
+
+        private URI computeConnectPath(URI connectURI) {
+            URI acceptURI = acceptSession.getServicePath();
+            URI requestURI = acceptSession.getRequestURI();
+            URI relativeURI = acceptURI.relativize(requestURI);
+
+            boolean hasExtraPath = relativeURI.getPath().length() > 0;
+            boolean requestEndsInSlash = requestURI.getPath().endsWith("/");
+            String connectPath = connectURI.getPath();
+            if (hasExtraPath || requestEndsInSlash) {
```

Comment:
- hasExtraPath - checks for path in request ( with or without query and/or fragments )
- requestEndsInSlash - covers the scenario where the request has no path and no query.